### PR TITLE
Created Workshops Page

### LINKED ISSUE #781: [NG] Migrate the site data for the Workshops page (Resources section)
**Description**

Currently, the following page on the Resources section on the [layer5-ng](https://layer5ng.netlify.app/) site points to a 404 error:

[Service Mesh Workshops](https://layer5ng.netlify.app/learn/workshops)

Migrate the content from the Layer5.io site. The data can be sourced from the following page on the [layer5.io](https://layer5.io/) site:

[Service Mesh Workshops](https://layer5.io/workshops)

---
**Contributor Resources**
The `layer5` repo contains two websites. The current generation and the next-generation of the layer5.io site.

If the `layer5-ng` label is absent on this issue, then this issue pertains to the current generation of the layer5.io website, which uses Jekyll and GitHub Pages. Site content is found under the [`master` branch](https://github.com/layer5io/layer5/tree/master).
- See [`layer5` contributing instructions](https://github.com/layer5io/layer5/blob/master/CONTRIBUTING.md)

If the `layer5-ng` label is present on this issue, then this issue pertains to the next-generation of the layer5.io website, which uses Gatsby, Strapi, and GitHub Pages. Site content is found under the [`layer5-ng` branch](https://github.com/layer5io/layer5/tree/layer5-ng).
- See [`layer5-ng` contributing instructions](https://github.com/layer5io/layer5/blob/layer5-ng/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by kishore007k**
Signed-off-by: kishore007k <kishorekumarstar07@gmail.com>

**Description**

This PR fixes #781 

**Notes for Reviewers**

@Nikhil-Ladha @Tanuj22 @Jashpatel1 

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by Jashpatel1** [comment]
@Tanuj22 is this an issue with netlify, as the preview bot is not up yet.

**Comment by kishore007k** [comment]
The site is working fine on my local machine. I guess the Netlify is causing the issue.

**Comment by Tanuj22** [comment]
Oh, it slipped my mind. We had to stop netlify build as we reached 75% of this month 

> @Tanuj22 is this an issue with github-actions or netlify? as the preview bot is not up yet.

**Comment by Tanuj22** [comment]
@kishore007k actually the data the is not so for each workshop what we want to create individual pages for them. We need a way to show it is the card itself. 
- Like this maybe - https://www.docker.com/events. But the problem in that would be we have varying amounts of details on the cards.  
- We can use a model to show the details on the hover. But maybe that won't we a good experience on mobile. 

So I think the best option is to go for the layout I suggested on slack. Your thought @kishore007k ?

**Comment by Tanuj22** [comment]
Just for reference, this is the current design - 
Screenshot from 2020-10-11 16-00-34

**Comment by kishore007k** [comment]
@Tanuj22 @Nikhil-Ladha @Jashpatel1 
Actually, the example that you suggested in the slack was good.

But I think It would be better if I create a  card that spans for full column and 2 rows something like this.
Capture9

It would contain the workshop title, image, some information about the workshop, and the see more (or) read more link.
When the user clicks on the Image or the Link it will take the user to that specific workshop page.

**Comment by Nikhil-Ladha** [comment]
Means, you are saying that on clicking the `know more` or any such button, a block will pop out beneath it occupying the whole width of the parent container, and then it will list the information, right?

**Comment by Nikhil-Ladha** [comment]
https://mdbootstrap.com/plugins/jquery/extended-cards/

Look at the `Gradient cards` design, a bit down on the page.
Having something like that would be best I suppose

**Comment by kishore007k** [comment]
> Means, you are saying that on clicking the `know more` or any such button, a block will pop out beneath it occupying the whole width of the parent container, and then it will list the information, right?

@Nikhil-Ladha  Yes something like that but it pops up in front of all the components taking up 80% of the total screen size. When the user clicks on the empty space or the close (or) return btn the popup will get closed.

**Comment by kishore007k** [comment]
> https://mdbootstrap.com/plugins/jquery/extended-cards/
> 
> Look at the `Gradient cards` design, a bit down on the page.
> Having something like that would be best I suppose

This looks great.

**Comment by Nikhil-Ladha** [comment]
> > https://mdbootstrap.com/plugins/jquery/extended-cards/
> > Look at the `Gradient cards` design, a bit down on the page.
> > Having something like that would be best I suppose
> 
> This looks great.

Plan to have something like this...if possible 🙂

**Comment by kishore007k** [comment]
@Nikhil-Ladha @Tanuj22 @Jashpatel1 
Let me add some screenshots for you guys to take a look. If you wonder about the site layout. I will also add a video in the slack please check that too for a better look.
Capture18
Capture19
Capture20

**Comment by kishore007k** [comment]
Oh It doesn't show the preview, Is this an issue or did they changed the settings like this?
Anyway, I'll send the screenshot and video in slack.

**Review by Jashpatel1 — APPROVED** [review]
We can move ahead with this view for now, and discuss about it in websites meeting.

**Comment by Nikhil-Ladha** [comment]
How about adding icons beside the buttons for slack, youtube and all.

**Comment by kishore007k** [comment]
@Nikhil-Ladha Yeah that would be nice.

**Comment by Nikhil-Ladha** [comment]
> @Nikhil-Ladha Yeah that would be nice.

Then, if you can please add them too. We would be good to go here.

**Comment by kishore007k** [comment]
Done @Nikhil-Ladha 
Capture21

**Comment by Nikhil-Ladha** [comment]
Better to have the left buttons inline with the `Show less/ Show more` button.

**Comment by kishore007k** [comment]
@Nikhil-Ladha Yeah that would be nice But in some cards, there was something like `upcoming...` when I keep the buttons inline with the `Show More/ Show Less` it gets interfered with.
Capture22
Do you want all these to be `inline`?

**Comment by Nikhil-Ladha** [comment]
See if we are showing`upcoming` in the page, then it doesn't makes any sense to show the slack, video or url buttons. As, logically they have not yet been created. So, what you do is check if there is something like `upcoming` field or property there just display that text and not rhe buttons. Otherwise, show the buttons.

**Comment by kishore007k** [comment]
> See if we are showing`upcoming` on the page, then it doesn't make any sense to show the slack, video, or URL buttons. As logically they have not yet been created. So, what you do is check if there is something like `upcoming` field or property there just display that text and not the buttons. Otherwise, show the buttons.

This is the current state of the site.
Capture23
Capture24
@Nikhil-Ladha

**Review by Nikhil-Ladha — APPROVED** [review]
Great work here @kishore007k . Really impressive :clap:

File: src/templates/workshop-single.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+// import React from "react"
```
Comment (Nikhil-Ladha) [inline src/templates/workshop-single.js:1]: I think this file is not needed. Can it be removed ?
@kishore007k , @Tanuj22 @Jashpatel1 ?

File: src/components/Workshop-Card/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+                </div>
+            </div>
+        </div>
+    </WorkshopCardWrapper>
+    )
+}
+export default WorkshopCardContent
+
+
+{/* <div className="btn-and-status">
```
Comment (Nikhil-Ladha) [inline src/components/Workshop-Card/index.js:27]: Remove the commented code

File: src/components/Workshop-Card/WorkshopsCardWrapper.style.js [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+    padding-left: 10px;
+    text-align: start;
+  }
+
+  @media only screen and (max-width: 480px) {
+  }
+`
+export default WorkshopCardWrapper
+
+// const WorkshopCardWrapper = styled.div`
```
Comment (Nikhil-Ladha) [inline src/components/Workshop-Card/WorkshopsCardWrapper.style.js:77]: Remove the commented code.

**Review by Nikhil-Ladha — CHANGES_REQUESTED** [review]

File: src/components/Workshop-Card/WorkshopsCardWrapper.style.js (hunk 3 above)
Comment (kishore007k) [reply src/components/Workshop-Card/WorkshopsCardWrapper.style.js:77]: Yeah I'll do all the clean up in this commit

File: gatsby-node.js [hunk 4]
```diff
@@ -61,6 +61,10 @@ exports.createPages = async ({ actions, graphql, reporter }) => {
       'src/templates/member-single.js'
   );
 
+  const WorkshopPostTemplate = path.resolve(
```
Comment (Tanuj22) [inline gatsby-node.js:64]: This can be removed

File: gatsby-node.js [hunk 5]
```diff
@@ -128,6 +132,10 @@ exports.createPages = async ({ actions, graphql, reporter }) => {
       node => node.fields.collection === `careers`
   );
 
+  const workshops = allNodes.filter(
```
Comment (Tanuj22) [inline gatsby-node.js:135]: This can be removed

File: gatsby-node.js [hunk 6]
```diff
@@ -221,6 +229,16 @@ exports.createPages = async ({ actions, graphql, reporter }) => {
     })
   });
 
+  workshops.forEach(singleWorkshop => {
```
Comment (Tanuj22) [inline gatsby-node.js:232]: This can be removed

File: src/pages/workshops.js [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+                ...GatsbyImageSharpFluid_withWebp
+              }
+            }
+            extension
+            publicURL
+          }
+        }
+        fields {
+          slug
+          collection
```
Comment (Tanuj22) [inline src/pages/workshops.js:47]: `collection` can be removed

File: src/sections/Workshop-grid/index.js [hunk 8]
```diff
@@ -0,0 +1,85 @@
+import React, { useState } from "react";
+import { MDXRenderer } from "gatsby-plugin-mdx";
+import { Container, Row, Col } from "../../reusecore/Layout";
+import WorkshopCard from "../../components/Workshop-Card";
+
+import { WorkshopPageWrapper } from "./WorkshopsGrid.style";
+
+// Social Icons
+import Slack from "../../images/socialIcons/slack_color.png"
```
Comment (Tanuj22) [inline src/sections/Workshop-grid/index.js:9]: Instead of using these icons. Use icons from `react-icons` .https://react-icons.github.io/react-icons/ .

Comment (kishore007k) [reply src/sections/Workshop-grid/index.js:9]: Yeah, I could but It won't have any color. Is that ok?

Comment (Tanuj22) [reply src/sections/Workshop-grid/index.js:9]: hmm..it will difficult to color the slack logo. Leave as it is.

**Review by Tanuj22 — APPROVED** [review]
Really good work @kishore007k . Thanks for working on this :)

**Review by Nikhil-Ladha — APPROVED** [review]

**Pull request merged**